src/appfl/run_mpi.py

```diff
@@ -245,26 +240,36 @@ def run_client(
         cfg.validation = False
         test_dataloader = None
 
-    clients = [
-        eval(cfg.fed.clientname)(
-            cid,
-            weight[cid],
-            copy.deepcopy(model),
-            loss_fn,
-            DataLoader(
-                train_data[cid],
-                num_workers=cfg.num_workers,
-                batch_size=batchsize[cid],
-                shuffle=cfg.train_data_shuffle,
-                pin_memory=True,
-            ),
-            cfg,
-            outfile[cid],
-            test_dataloader,
-            **cfg.fed.args,
+    clientpergpu = math.ceil(num_clients/cfg.num_gpu)    
+    clients = []
+    for _, cid in enumerate(num_client_groups[comm_rank - 1]):
+        ## We assume to have as many GPUs as the number of MPI processes.
+        if "cuda" in cfg.device:
+            gpuindex = int(math.floor(cid/clientpergpu))            
+            device = f"cuda:{gpuindex}"
```

Comment:
I think `gpuindex` may better be a configuration parameter that can be set for each client. For example, we may better set this parameter in the example file.

Reply:
I'm not sure if I understand correctly your comment. Does it mean that the user should set what GPU to assign for each client or set the number of clients for each GPU? 
I believe it would be nice to have a feature that distributes the clients to all available GPUs to utilize all the resources in the machine.